pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -154,6 +154,8 @@ def restore_datamodule(self) -> None:
         datamodule = self.trainer.datamodule
         if datamodule is not None:
             datamodule.on_load_checkpoint(self._loaded_checkpoint)
+            if datamodule.__class__.__name__ in self._loaded_checkpoint:
```

Comment:
https://github.com/PyTorchLightning/pytorch-lightning/blob/86b177ebe5427725b35fde1a8808a7b59b8a277a/pytorch_lightning/callbacks/base.py#L43 

let's use qualname here too for consistency